x/internal/outline/outline_packet_proxy.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package outline
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/Jigsaw-Code/outline-sdk/network"
+	"github.com/Jigsaw-Code/outline-sdk/network/dnstruncate"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/Jigsaw-Code/outline-sdk/transport/shadowsocks"
+	"github.com/Jigsaw-Code/outline-sdk/x/connectivity"
+)
+
+type OutlinePacketProxy interface {
+	network.PacketProxy
+
+	TestConnectivityAndRefresh(resolver, domain string) error
+}
+
+type outlinePacketProxy struct {
+	network.DelegatePacketProxy
+
+	remotePktListener transport.PacketListener // this will be used in connectivity test
+	remote, fallback  network.PacketProxy
+}
+
+var _ OutlinePacketProxy = (*outlinePacketProxy)(nil)
+
+func NewOutlinePacketProxy(config *SessionConfig) (opp OutlinePacketProxy, err error) {
```

Comment:
Take an access key instead. Same for NewOutlinePacketListener.

The config we actually use might be ok, but it's more work for the user.